### PR TITLE
Generate session keys in the Orchestrator

### PR DESCRIPTION
**Pull request by ipetr0v**
This PR adds the ability to generate session keys in the Orchestrator and request them form the trusted application.

### CODE REVIEW

**Comment by ipetr0v** [comment]
@tiziano88 I tried using Proto dependencies like in https://github.com/project-oak/oak/pull/4211, but get the following error:
```
error[E0433]: failed to resolve: could not find `crypto` in `super`
  --> /usr/local/google/home/ivanpetrov/src/oak/target/x86_64-unknown-linux-gnu/debug/build/oak_containers_orchestrator_client-eb486495a677e1b0/out/oak.containers.rs:42:48
   |
42 |     pub context: ::core::option::Option<super::crypto::v1::CryptoContext>,
   |                                                ^^^^^^ could not find `crypto` in `super`
```

File: oak_containers_hello_world_trusted_app/src/orchestrator_client.rs [hunk 1]
```diff
@@ -45,8 +35,9 @@ static IGNORED_ENDPOINT_URI: &str = "file://[::]:0";
 const IPC_SOCKET: &str = "/oak_utils/orchestrator_ipc";
 
 /// Utility struct used to interface with the launcher
+#[derive(Clone)]
 pub struct OrchestratorClient {
-    inner: GrpcOrchestratorClient<tonic::transport::channel::Channel>,
+    inner: Arc<Mutex<GrpcOrchestratorClient<tonic::transport::channel::Channel>>>,
```
Comment (conradgrobler) [inline oak_containers_hello_world_trusted_app/src/orchestrator_client.rs:40]: Why is this needed? The Arc shouldn't be needed becase Tonic clients are cheap to clone. The mutex should not be needed since invoking any of the methods requires a mutable reference to self, so there will be one reference to a client instance at a time.

File: oak_containers_orchestrator/src/ipc_server.rs [hunk 2]
```diff
@@ -13,32 +13,23 @@
...
+use oak_containers_orchestrator_client::LauncherClient;
+use oak_crypto::encryptor::{EncryptionKeyProvider, RecipientContextGenerator};
+use oak_remote_attestation::attester::Attester;
+use std::sync::Arc;
 use tokio::net::UnixListener;
 use tokio_stream::wrappers::UnixListenerStream;
 use tonic::{transport::Server, Request, Response};
 
 pub struct ServiceImplementation {
+    attester: Arc<Attester>,
```
Comment (conradgrobler) [inline oak_containers_orchestrator/src/ipc_server.rs:31]: Why are Arcs needed here?

Comment (ipetr0v) [reply oak_containers_orchestrator/src/ipc_server.rs]: Removed

File: oak_containers_hello_world_trusted_app/src/orchestrator_client.rs (hunk 1 above)
Comment (ipetr0v) [reply oak_containers_hello_world_trusted_app/src/orchestrator_client.rs]: Done

**Review by conradgrobler — APPROVED** [review]

**Comment by ipetr0v** [comment]
> @tiziano88 I tried using Proto dependencies like in #4211, but get the following error:
> 
> ```
> error[E0433]: failed to resolve: could not find `crypto` in `super`
>   --> /usr/local/google/home/ivanpetrov/src/oak/target/x86_64-unknown-linux-gnu/debug/build/oak_containers_orchestrator_client-eb486495a677e1b0/out/oak.containers.rs:42:48
>    |
> 42 |     pub context: ::core::option::Option<super::crypto::v1::CryptoContext>,
>    |                                                ^^^^^^ could not find `crypto` in `super`
> ```

The error was coming from `oak_containers_orchestrator_client` crate. Added `pub use oak_crypto::proto::oak::crypto;` there and now it's fixed.
